### DISCUSSION TITLE
Action after SpreadSheet FULLY saved

### DISCUSSION THREAD

**Discussion by rvbz-swq** [PHPOffice/PhpSpreadsheet#2553] [Q&A]
I'm wondering if there's a proper way to know if the spreadsheet file has been fully and completely saved. 
At the moment, I have a XLSX writer:

`$writer->save($docPath);`

After this piece of code I'm writing the the file path into my DB so that I can show it to the user for download purposes, the issue that I'm currently having is that apparently the record on my DB is been shown before the spreadsheet file has finished writing all the information into it, thus some users download the file with just half of the data in it (if they click the download button too early).

Any recommendations for this issue?

**Comment by MarkBaker** [comment]
The call to `save()` is synchronous, so the file should be fully saved when that call completes. However operating systems don't always write data to disk immediately; but that's outside the control of PHP until version 8.1, which introduced the `fsync()` function that forces the OS to write to disk

**Comment by MarkBaker** [comment]
However, if only part of the data is in an Xlsx file, it suggests that not all of the data has been included in the file; a partly-written Xlsx file would be corrupted, and wouldn't open in Excel

**Reply by rvbz-swq** [reply]
That's exactly what I thought, I couldn't see any other way to save the file but synchronous, and then I thought what you mention, a corrupted file has been created, but surprisingly enough the file is not corrupted, it can be opened and all, it just does not have all the data in it, really weird
